docs/getting-started/openshift/index.md

```diff
@@ -0,0 +1,212 @@
+
+Issue the certificate for the principal's resource proxy:
+The principal's resource proxy needs to be reachable by your Argo CD API server (argocd-server), which usually runs in the same cluster as the principal. Typically, this service should not be reachable outside the cluster and is exposed using a Kubernetes Service.
+
+```
+argocd-agentctl pki issue resource-proxy \
+    --principal-context <control plane context> \
+    --ip <ip addresses of resource proxy> \
+    --dns <dns names of principal>
+```
+
+Deploy principal on hub cluster using argocd-operator/gitops-operator using Argo CD CR given below
+
+```
+apiVersion: argoproj.io/v1beta1
+kind: ArgoCD
+metadata:
+  name: argocd
+spec:
+  controller:
+    enabled: false
+  argoCDAgent:
+    principal:
+      enabled: true
+      allowedNamespaces: 
+        - "*"
+      jwtAllowGenerate: true
+      auth: "mtls:CN=([^,]+)"
+      logLevel: "trace"
+      image: "ghcr.io/argoproj-labs/argocd-agent/argocd-agent:latest"
+  sourceNamespaces:
+    - "agent-managed"
+    - "agent-autonomous"  					
+```
+
+The above CR should create all the necessary resource for Argo CD as well as argocd-agent principal in argocd namespace.
+
+Create argocd-redis secret, because principal looks for it to fetch redis authentication details.
+
+```
+oc create secret generic argocd-redis -n argocd --from-literal=auth="$(oc get secret argocd-redis-initial-password -n argocd -o jsonpath='{.data.admin\.password}' | base64 -d)"
+```
+
+## Setting up agent workload cluster 
+
+### Configure Argo-CD for Agent 
+ 
+Argo CD instance on Agent cluster
+
+Creating Argo CD instance for Workload/spoke cluster.
+```
+  apiVersion: argoproj.io/v1beta1
+  kind: ArgoCD
+  metadata:
+    name: argocd
+  spec:
+    server:
+      enabled: false
+```
+
+### Configure Agent in managed mode
+
+Before installing agent resources create 
+- a TLS secret containing the issued certificate for agent
+
+Create the PKI on the agent:
+Run this command while connected to principal
+```
+argocd-agentctl pki issue agent <agent-name>  --principal-context <principal context> --agent-context <workload context> --agent-namespace <workload namespace> --upsert
+```
+
+Apply the installation manifests for Argo CD-agent agent
+```
+oc apply -n $(workload-namespace) -k 'https://github.com/argoproj-labs/argocd-agent/install/kubernetes/agent?ref=main'
+```
+This should create all the required agent related resources.
+
+Update the configMap with name `argocd-agent-params`  with parameters related to agent.mode,agent.creds, agent.namespace, agent.server.address.	
+```
+  agent.keep-alive-ping-interval: 50s
+  agent.mode: managed
+  agent.creds: mtls:any
+  agent.tls.client.insecure: "false"
+  agent.tls.root-ca-path: "/app/config/tls/ca.crt"
+  agent.tls.client.cert-path: "/app/config/tls/tls.crt"
+  agent.tls.client.key-path: "/app/config/tls/tls.key"
```

Comment:
Using the path doesn't seem to work because the secrets are not mounted in the deployment yaml properly.
The easiest way to set them via secret name is already done below in `agent.tls.root-ca-secret-name` and `agent.tls.secret-name`. I think these three path values should be removed.

Reply:
Sounds good.